src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -374,9 +374,15 @@ static TestResult getTestResult(Job<?, ?> originProject, Run<?, ?> b, TaskListen
                 if (tra != null) {
                     Object o = tra.getResult();
                     if (o instanceof TestResult) {
-                        listener.getLogger().printf("Using build %s as reference%n", ModelHyperlinkNote.encodeTo('/' + b.getUrl(), originProject != b.getParent() ? b.getFullDisplayName() : b.getDisplayName()));
-                        result = (TestResult) o;
-                        break;
+                        TestResult tr = (TestResult) o;
+                        String hyperlink = ModelHyperlinkNote.encodeTo('/' + b.getUrl(), originProject != b.getParent() ? b.getFullDisplayName() : b.getDisplayName());
+                        if (tr.getTotalCount() == 0) {
+                            listener.getLogger().printf("Build %s has no loadable test results (supposed count %d), skipping%n", tra.getTotalCount(), hyperlink);
```

Comment:
Note persisted `Integer totalCount`, as opposed to actually loading the results and counting.